include/rdma/fi_domain.h

```diff
@@ -183,6 +183,21 @@ struct fi_mr_modify {
 	struct fi_mr_attr	attr;
 };
 
+struct fi_mr_alloc_keys {
+	size_t		count;
+	uint64_t	*keys;
```

Comment:
This assumes that the provider must support 64-bit keys, which may not be the case.  It isn't for all existing providers.